### PR TITLE
LOG-5604: Add input validation

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* adds input validation
* Drops validations that should be addressed by API annotations

### Links
* https://issues.redhat.com/browse/LOG-5604

cc @vparfonov @Clee2691

### CODE REVIEW

File: internal/validations/observability/inputs/application.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+			message = fmt.Sprintf(message, args...)
+		}
+		return internalobs.NewCondition(obs.ValidationCondition,
+			metav1.ConditionTrue,
+			reason,
+			message,
+		)
+	}
+
+	if spec.Application == nil {
```
Comment (vparfonov) [inline internal/validations/observability/inputs/application.go:32]: this behavior not agreed with `//+kubebuilder:validation:Optional` in `InputSpec`

File: internal/validations/observability/inputs/receivers.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		return []metav1.Condition{
+			internalobs.NewCondition(obs.ValidationCondition,
+				metav1.ConditionTrue,
+				obs.ReasonValidationFailure,
+				message,
+			),
+		}
+	}
+
+	if spec.Receiver == nil {
```
Comment (vparfonov) [inline internal/validations/observability/inputs/receivers.go:25]:
```suggestion
	if spec.Receiver.Type == obs.ReceiverTypeHTTP {
    if spec.Receiver.HTTP == nil {
        return newCond(fmt.Sprintf("%s has nil HTTP receiver spec", spec.Name))
    }

    if spec.Receiver.HTTP.Format == "" {
        return newCond(fmt.Sprintf("%s does not specify a format", spec.Name))
    }
 }
```

File: internal/validations/observability/inputs/application.go (hunk 1 above)
Comment (jcantrill) [reply internal/validations/observability/inputs/application.go:32]: @vparfonov i don't understand what you are expressing. AFAIK there is no way to require a type and enforce the associated spec

**Review by jcantrill — COMMENTED** [review]
/hold cancel

**Pull request merged**